src/iperf_udp.c

```diff
@@ -564,17 +566,35 @@ iperf_udp_connect(struct iperf_test *test)
      * Write a datagram to the UDP stream to let the server know we're here.
      * The server learns our address by obtaining its peer's address.
      */
-    buf = 123456789;		/* this can be pretty much anything */
+    buf = UDP_CONNECT_MSG;
+    if (test->debug) {
+        printf("Sending Connect message to Sockt %d\n", s);
+    }
     if (write(s, &buf, sizeof(buf)) < 0) {
         // XXX: Should this be changed to IESTREAMCONNECT?
         i_errno = IESTREAMWRITE;
         return -1;
     }
 
     /*
-     * Wait until the server replies back to us.
+     * Wait until the server replies back to us with the "accept" response.
      */
-    if ((sz = recv(s, &buf, sizeof(buf), 0)) < 0) {
+    i = 0;
+    max_len_wait_for_reply = sizeof(buf);
+    if (test->reverse) /* In reverse mode allow few packets to have the "accept" response - to handle out of order packets */
+        max_len_wait_for_reply += MAX_REVERSE_OUT_OF_ORDER_PACKETS * test->settings->blksize;
+    do {
+        if ((sz = recv(s, &buf, sizeof(buf), 0)) < 0) {
+            i_errno = IESTREAMREAD;
+            return -1;
+        }
+        if (test->debug) {
+            printf("Connect received for Socket %d, sz=%d, buf=%x, i=%d, max_len_wait_for_reply=%d\n", s, sz, buf, i, max_len_wait_for_reply);
+        }
+        i += sz;
+    } while (buf != UDP_CONNECT_REPLY && i < max_len_wait_for_reply);
```

Comment:
Done.  Forgot about the backward compatibility ...

Note that there is a minor potential issue with checking for both the old and new connect reply values.  Potentially, the first packet that is received instead of the connect reply (because of the out of order) can contain the expected reply value on a 4 bytes boundary.  Checking for both values doubles the chance that that will happen.  On the other hand, practically that can probably happen only when sending a file contents.  Since out of order is rare, and this only about the first packet, checking for both values is probably o.k.  If not, then only the legacy value should be used.